modules/rostests/apitests/ntdll/RtlCriticalSection.c

```diff
@@ -0,0 +1,374 @@
+                ok(CritSect.DebugInfo->ProcessLocksList.Flink != NULL, "Flink is NULL\n");
+                ok(CritSect.DebugInfo->ProcessLocksList.Blink != NULL, "Blink is NULL\n");
+                if ((CritSect.DebugInfo->ProcessLocksList.Flink != NULL) &&
+                    (CritSect.DebugInfo->ProcessLocksList.Blink != NULL))
+                {
+                    ok_ptr(CritSect.DebugInfo->ProcessLocksList.Flink->Blink,
+                        &CritSect.DebugInfo->ProcessLocksList);
+                    ok_ptr(CritSect.DebugInfo->ProcessLocksList.Blink->Flink,
+                        &CritSect.DebugInfo->ProcessLocksList);
+                }
+                ok_long(CritSect.DebugInfo->EntryCount, 0);
+                ok_long(CritSect.DebugInfo->ContentionCount, 0);
+                ok_long(CritSect.DebugInfo->Flags, 0);
+                ok_int(CritSect.DebugInfo->SpareWORD, 0);
+            }
+        }
+    }
+    else
+    {
+        skip("RtlInitializeCriticalSectionEx not available.\n");
+    }
+}
+
+static
+DWORD
+WINAPI
+ThreadProc1(
+    _In_ LPVOID lpParameter)
+{
+    printf("ThreadProc1 starting\n");
+    RtlEnterCriticalSection(&CritSect);
+
+    SetEvent(hEventThread1Ready);
+
+    printf("ThreadProc1 waiting\n");
+    WaitForSingleObject(hEventThread1Cont, INFINITE);
+    printf("ThreadProc1 returned from wait\n");
+
+    RtlLeaveCriticalSection(&CritSect);
+
+    return 0;
+}
+
+static
+DWORD
+WINAPI
+ThreadProc2(
+    _In_ LPVOID lpParameter)
+{
+    printf("ThreadProc2 starting\n");
+    RtlEnterCriticalSection(&CritSect);
+
+    SetEvent(hEventThread2Ready);
+
+    printf("ThreadProc2 waiting\n");
+    WaitForSingleObject(hEventThread2Cont, INFINITE);
+    printf("ThreadProc2 returned from wait\n");
+
+    RtlLeaveCriticalSection(&CritSect);
+
+    return 0;
+}
+
+static
+void
+Test_Acquire(void)
+{
+    DWORD dwThreadId1, dwThreadId2;
+
+    RtlInitializeCriticalSection(&CritSect);
+
+    // Acquire once
+    RtlEnterCriticalSection(&CritSect);
+    ok_long(CritSect.LockCount, -2);
+    ok_long(CritSect.RecursionCount, 1);
+    ok_ptr(CritSect.OwningThread, UlongToHandle(GetCurrentThreadId()));
+    ok_ptr(CritSect.LockSemaphore, NULL);
+    //ok_size_t(CritSect.SpinCount, 0x20007d0);
+
+    // Acquire recursively
+    RtlEnterCriticalSection(&CritSect);
+    ok_long(CritSect.LockCount, -2);
+    ok_long(CritSect.RecursionCount, 2);
+    ok_ptr(CritSect.OwningThread, UlongToHandle(GetCurrentThreadId()));
+    ok_ptr(CritSect.LockSemaphore, NULL);
+    //ok_size_t(CritSect.SpinCount, 0x20007d0);
+
+    hEventThread1Ready = CreateEvent(NULL, TRUE, FALSE, NULL);
+    hEventThread1Cont = CreateEvent(NULL, TRUE, FALSE, NULL);
+    hEventThread2Ready = CreateEvent(NULL, TRUE, FALSE, NULL);
+    hEventThread2Cont = CreateEvent(NULL, TRUE, FALSE, NULL);
+
+    // Create thread 1 and wait to it time to try to acquire the critical section
+    CreateThread(NULL, 0, ThreadProc1, NULL, 0, &dwThreadId1);
+    Sleep(10);
```

Comment:
This seems short. Perhaps poll on `LockCount` with a timeout, to avoid flakiness?